### PR TITLE
Added SNAPSHOT flag when git repository is dirty

### LINKED ISSUE #35: Unique version should output a SNAPSHOT if the git repo is dirty
It's fairly misleading if a JAR is generated with a commish embedded in the name/version when the git repository is dirty. It should instead generate a `SNAPSHOT` version instead (or something which indicates its not an actual git commit - maybe a hash of the file, etc).

### PR DESCRIPTION
**Pull request by laurencer**
Fixes #35 and #23

### CODE REVIEW

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/GitInfo.scala [hunk 1]
```diff
@@ -19,13 +19,45 @@ import java.io.File
...
+
+case object NotAGitRepository extends GitStatus
+
+case class Clean(hash: String) extends GitStatus
+
+case class Dirty(hash: String) extends GitStatus
+
 object GitInfo {
-  def commish(root: File) =
+
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/GitInfo.scala]: You added an unnecessary empty line here.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/GitInfo.scala [hunk 2]
```diff
@@ -19,13 +19,45 @@ import java.io.File
...
+      NotAGitRepository
+    }
+  }
+
+  def isGitRepository(root: File): Boolean =
+    Process("git status", Some(root)).! == 0
+
+  def isGitRepositoryDirty(root: File): Boolean =
+    Process("git status --porcelain", Some(root)).lines.nonEmpty
+
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/GitInfo.scala]: Same.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/UniqueVersionPlugin.scala [hunk 3]
```diff
@@ -20,8 +20,8 @@ import sbt._, Keys._
 object UniqueVersionPlugin extends Plugin {
   import Times._
   import GitInfo._
-
+
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/version/UniqueVersionPlugin.scala]: Trailing whitespace.

**Comment by stephanh** [comment]
Very nice! :+1:

**Comment by laurencer** [comment]
@stephanh Fixed your comments - good to merge?

**Comment by stephanh** [comment]
@laurencer did you push your changes?

**Comment by laurencer** [comment]
@stephanh done!

**Pull request merged**